src/server/debug/server/database_dump.go

```diff
@@ -1,38 +1,41 @@
 package server
 
 import (
-	"archive/tar"
 	"bufio"
 	"context"
 	"database/sql"
 	"encoding/json"
 	"fmt"
 	"io"
+	"path/filepath"
 	"strings"
 	"time"
 
 	"github.com/jmoiron/sqlx"
+	"go.uber.org/multierr"
 
 	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
 	"github.com/pachyderm/pachyderm/v2/src/internal/log"
 	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
 )
 
-func (s *debugServer) collectDatabaseDump(ctx context.Context, tw *tar.Writer, prefix ...string) (retErr error) {
+func (s *debugServer) collectDatabaseDump(ctx context.Context, dir string, rp reportProgressFunc) (retErr error) {
 	defer log.Span(ctx, "collectDatabaseDump")(log.Errorp(&retErr))
 	ctxWithTimeout, cancel := context.WithTimeout(ctx, 1*time.Minute)
 	defer cancel()
+	rp(ctx, 0, 100)
 	var errs error
-	if err := s.collectDatabaseStats(ctxWithTimeout, tw, prefix...); err != nil {
-		errors.JoinInto(&errs, errors.Wrap(err, "collectDatabaseStats"))
+	if err := s.collectDatabaseStats(ctxWithTimeout, dir); err != nil {
+		multierr.AppendInto(&errs, errors.Wrap(err, "collectDatabaseStats"))
```

Comment:
errors.JoinInto is pachyderm/pachyderm's version of multierr.AppendInto.  You'll notice that the linter gets mad about importing multierr now.